### ISSUE TITLE
Samba service not starting after installing primary DC

### ISSUE THREAD

**Issue by Carbon401** [fumatchu/RADS#1]
I've started the script to install a primary DC on a proxmox Rocky9 VM. It's a fresh minimal install - I only selected a static IP and FQDN in the network section. The script finsihed but failed the testing part in the end. I've noticed that samba service is not running and can not be started. I also noticed that there's no current /etc/samba/smb.conf file. Any help debugging this? Thanks

```
[root@dc ~]# systemctl restart samba
Job for samba.service failed because the control process exited with error code.
See "systemctl status samba.service" and "journalctl -xeu samba.service" for details.
[root@dc ~]# systemctl status samba
× samba.service - Samba AD Daemon
     Loaded: loaded (/usr/lib/systemd/system/samba.service; enabled; preset: disabled)
     Active: failed (Result: exit-code) since Tue 2024-07-09 10:32:17 EEST; 2s ago
       Docs: man:samba(8)
             man:samba(7)
             man:smb.conf(5)
    Process: 1523 ExecStart=/usr/sbin/samba --foreground --no-process-group $SAMBAOPTIONS (code=exited, status=1/FAILURE)
   Main PID: 1523 (code=exited, status=1/FAILURE)
        CPU: 37ms

Jul 09 10:32:17 dc.***.lan systemd[1]: Starting Samba AD Daemon...
Jul 09 10:32:17 dc.***.lan systemd[1]: samba.service: Main process exited, code=exited, status=1/FAILURE
Jul 09 10:32:17 dc.***.lan systemd[1]: samba.service: Failed with result 'exit-code'.
Jul 09 10:32:17 dc.***.lan systemd[1]: Failed to start Samba AD Daemon.
[root@dc ~]# ls /etc/samba/
lmhosts  smb.bak.orig  smb.conf.example
```

**Comment by fumatchu** [comment]
I had another individual updating their DNS incorrectly.. the script will take care of DNS. On the install, just point the server to another DNS server other than itself. When you setup your network stack, did you use an external DNS server?

**Comment by Carbon401** [comment]
Hello, thanks for helping me with this.
`nmcli | grep "connected to" | cut -c22-` output is `ns18`
but the correct interface name is `ens18`.
Probably a better strategy is to use `cut -d " " -f4` instead of `cut -c22-`

The raw output of `nmcli | grep "connected to"` is `ens18: connected to ens18`.

**Comment by fumatchu** [comment]
OK, so that's the issue. 
I was wondering if that was going to happen. thank you for the input.. let me update this and thank you for the recommendation. I will let you know when this is updated

**Comment by fumatchu** [comment]
I am running  a test to make sure everything is good.. give me about 30 minutes to make sure all is good and I will republish.. Thank you again for your input

**Comment by fumatchu** [comment]
OK, I updated the scripts.. Things look good.. Please let me know your results, and thank you again for being patient and your suggestion. I greatly appreciate it

**Comment by Carbon401** [comment]
Again no go.
`Can't load /etc/samba/smb.conf - run testparm to debug it`
/etc/samba/smb.conf is missing.

**Comment by fumatchu** [comment]
Would you have time to get on a Webex with me?  I would be interested to see your deployment as it runs

**Comment by fumatchu** [comment]
One more thought.. the AD password:
 https://wiki.samba.org/index.php/Setting_up_Samba_as_an_Active_Directory_Domain_Controller

<html>
<body>

Administrator password | --adminpass | Sets the domain administrator password. If the password does not match the complexity requirements, the provisioning fails. For details, see Microsoft TechNet: Passwords must meet complexity requirements.
-- | -- | --



</body>
</html>Administrator password	--adminpass	Sets the domain administrator password. If the password does not match the complexity requirements, the provisioning fails. For details, see [Microsoft TechNet: Passwords must meet complexity requirements](https://technet.microsoft.com/en-us/library/cc786468%28v=ws.10%29.aspx).

**Comment by Carbon401** [comment]
Thanks. It was the password complexity. Now it works. But maybe you should check the exit code and show the error so there's no guessing involved.

Also as a suggestion maybe it will be a good addition if you include this plugin with the cockpit installation: https://wiki.samba.org/index.php/GSOC_cockpit_samba_ad_dc

Again thanks for you work.

**Comment by fumatchu** [comment]
Thank you for your input. I will admit that I am learning as I have created this project. I do really appreciate all your input and suggestions and being patient as we worked through the challenges. I have also learned from this experience and I greatly appreciate it, to make the script better, which I will continue to mature.  
To answer your question, the script uses the Internal Samba DNS resolver. As far as I know, to this point, it only has one forwarder and that is to send everything to the upstream (DNS) forwarder that is in the smb.conf file.  There is the concept of the bind9_DLZ that is supported by Samba, that will allow you to do what you are asking. 

https://wiki.samba.org/index.php/BIND9_DLZ_DNS_Back_End

This is actually on my concept "todo list" and is something I would like to implement in the next 6 months. Currently, for work, I am working on a couple projects and hope to come back to this one in the fall. I have no intention of leaving this one "as-is". 

Again, I am grateful for your patience and your suggestions. Thank you, again!
